MiniAn/minian_main.py

```diff
@@ -0,0 +1,371 @@
+    cluster = LocalCluster(**minian_parameters.params_LocalCluster)
+    annt_plugin = TaskAnnotation()
+    cluster.scheduler.add_plugin(annt_plugin)
+    client = Client(cluster)
+
+    # MiniAn CNMF
+    intpath = os.path.join(minian_parameters.paths[defs.Parameters.Path.TMP_DIR], mn_defs.Text.Main.INTERMEDIATE)
+    subset = {"frame": slice(0, None)}
+
+    file_, chk, varr_ref = load_chunk(intpath, subset, minian_parameters)
+
+    varr_ref = preprocess(varr_ref, intpath, minian_parameters)
+
+    Y, Y_fm_chk, Y_hw_chk = correct_motion(varr_ref, intpath, chk, minian_parameters)
+
+    seeds_final, _ = initialize_seeds(Y_fm_chk, Y_hw_chk, minian_parameters)
+
+    A, C, C_chk, f, b = initialize_components(Y_hw_chk, Y_fm_chk, seeds_final, intpath, chk, minian_parameters)
+
+    A, C, C_chk, sn_spatial = cnmf1(Y_hw_chk, intpath, A, C, C_chk, Y_fm_chk, chk, minian_parameters)
+
+    A, C, AC, S, c0, b0 = cnmf2(Y_hw_chk, A, C, sn_spatial, intpath, C_chk, Y_fm_chk, chk, minian_parameters)
+
+    ### Save final results ###
+    print(mn_defs.Messages.Main.SAVING_FINAL, flush=True)
+    A = save_minian(A.rename("A"), **minian_parameters.params_save_minian)
+    C = save_minian(C.rename("C"), **minian_parameters.params_save_minian)
+    AC = save_minian(AC.rename("AC"), **minian_parameters.params_save_minian)
+    S = save_minian(S.rename("S"), **minian_parameters.params_save_minian)
+    c0 = save_minian(c0.rename("c0"), **minian_parameters.params_save_minian)
+    b0 = save_minian(b0.rename("b0"), **minian_parameters.params_save_minian)
+    b = save_minian(b.rename("b"), **minian_parameters.params_save_minian)
+    f = save_minian(f.rename("f"), **minian_parameters.params_save_minian)
+
+    ### Save results to doric file ###
+    print(mn_defs.Messages.Main.SAVING_TO_DORIC, flush=True)
+    # Get the path from the source data
+    data, driver, operation, series, sensor = minian_parameters.get_h5path_names()
+
+    # Get paramaters of the operation on source data
+    params_source_data = utils.load_attributes(file_, f"{data}/{driver}/{operation}")
+    # Get the attributes of the images stack
+    attrs = utils.load_attributes(file_, f"{minian_parameters.clean_h5path()}/{defs.DoricFile.Dataset.IMAGE_STACK}")
+    file_.close()
+
+    # Parameters
+    # Set only "Operations" for params_srouce_data
+    if defs.DoricFile.Attribute.OPERATION_NAME in params_source_data:
+        if defs.DoricFile.Attribute.OPERATIONS not in params_source_data:
+            params_source_data[defs.DoricFile.Attribute.OPERATIONS] = params_source_data[defs.DoricFile.Attribute.OPERATION_NAME]
+
+        del params_source_data[defs.DoricFile.Attribute.OPERATION_NAME]
+
+    if minian_parameters.parameters[defs.Parameters.danse.SPATIAL_DOWNSAMPLE] > 1:
+        minian_parameters.parameters[defs.DoricFile.Attribute.BINNING_FACTOR] = minian_parameters.parameters[defs.Parameters.danse.SPATIAL_DOWNSAMPLE]
+
+    mn_utils.save_minian_to_doric(
```

Comment:
I think the functions from minian_utilities can me just copied here, and the file deprecated

Reply:
Yeap good idea